docs(migrate-from-newrelic): add documentation on migrating from New Relic

tdabasinskas opened the pull request:
Fixes #1377

tdabasinskas committed:
chore(nav): update navigation menu

tdabasinskas committed:
feat(migrate-from-newrelic): add the main overview page

tdabasinskas committed:
feat(migrate-from-newrelic): add dashboards page

tdabasinskas committed:
feat(migrate-from-newrelic): add metrics page

tdabasinskas committed:
feat(migrate-from-newrelic): add logs page

tdabasinskas committed:
feat(migrate-from-newrelic): add traces page

tdabasinskas committed:
feat(migrate-from-newrelic): add alerts page

tdabasinskas committed:
chore(migrate-from-newrelic): do some minor formatting improvements

tdabasinskas committed:
chore(migrate-from-newrelic): rebuild

tdabasinskas committed:
Merge branch 'main' into migrate_newrelic

Calm-Rock commented on data/docs/migration/migrate-from-newrelic/alerts.mdx:
```diff
@@ -0,0 +1,94 @@
+---
+date: 2025-05-12
+id: migrate-from-newrelic-alerts
```
```suggestion
id: alerts
```

Calm-Rock commented on data/docs/migration/migrate-from-newrelic/alerts.mdx:
```diff
@@ -0,0 +1,94 @@
+
+## Using Terraform for Alert Management
+
+In case you define alert rules as code in your LGTM Stack (either as Prometheus rules or using Grafana Terraform provider), you can continue using the same approach while migrating to SigNoz, by utilizing the [SigNoz Terraform Provider](/docs/alerts-management/terraform-provider-signoz/).
```
```suggestion
In case you define alert rules as code in your LGTM Stack (either as Prometheus rules or using Grafana Terraform provider), you can continue using the same approach while migrating to SigNoz, by utilizing the [SigNoz Terraform Provider](/docs/alerts-management/terraform-provider-signoz/).
```

LGTM Stack? This is for New Relic, right?

Calm-Rock commented on data/docs/migration/migrate-from-newrelic/dashboards.mdx:
```diff
@@ -0,0 +1,122 @@
+---
+date: 2025-05-12
+id: migrate-from-newrelic-dashboards
```
```suggestion
id: dashboards
```

Calm-Rock commented on data/docs/migration/migrate-from-newrelic/logs.mdx:
```diff
@@ -0,0 +1,194 @@
+---
+date: 2025-05-12
+id: migrate-from-newrelic-logs
```
```suggestion
id: logs
```

Calm-Rock commented on data/docs/migration/migrate-from-newrelic/metrics.mdx:
```diff
@@ -0,0 +1,187 @@
+---
+date: 2025-05-12
+id: migrate-from-newrelic-metrics
```
```suggestion
id: metrics
```

Calm-Rock commented on data/docs/migration/migrate-from-newrelic/traces.mdx:
```diff
@@ -0,0 +1,130 @@
+---
+date: 2025-05-12
+id: migrate-from-newrelic-traces
```
```suggestion
id: traces
```

Calm-Rock commented on data/docs/migration/migrate-from-newrelic/traces.mdx:
```diff
@@ -0,0 +1,130 @@
+- [Ruby](/docs/instrumentation/ruby)
+- [.NET](/docs/instrumentation/dotnet)
+- [PHP](/docs/instrumentation/php)
+
```
```suggestion
Check this [document](https://signoz.io/docs/instrumentation/) for a list of languages and frameworks
```

Calm-Rock commented on data/docs/migration/migrate-from-newrelic/traces.mdx:
````diff
@@ -0,0 +1,130 @@
+     -jar your-application.jar
+```
+
+For more detailed instructions, see the [SigNoz Java Instrumentation Guide](/docs/instrumentation/java).
````
```suggestion
For more detailed instructions, see the [SigNoz Java Instrumentation document](/docs/instrumentation/java).
```

makeavish commented on data/docs/migration/migrate-from-newrelic/dashboards.mdx:
```diff
@@ -0,0 +1,122 @@
+
+| New Relic Dashboard Type    | SigNoz Template Equivalent                                                                                              | Description                                        |
+|-----------------------------|------------------------------------------------------------------------------------------------------------------------|----------------------------------------------------|
+| Infrastructure Monitoring   | [Host Metrics](https://github.com/SigNoz/dashboards/blob/main/hostmetrics/hostmetrics.json)                             | CPU, memory, disk, and network metrics for hosts   |
```
SigNoz also has infrastructure monitoring module now - https://signoz.io/docs/infrastructure-monitoring/overview/
So host metrics dashboard is optional. So is true for k8s monitoring.

tdabasinskas replied:
Nice! I've added a note under the table to highlight this. Let me know if that's good or you would like this to be put somehow differently.

makeavish commented on data/docs/migration/migrate-from-newrelic/metrics.mdx:
```diff
@@ -0,0 +1,187 @@
+
+For infrastructure metrics previously collected by New Relic Infrastructure:
+
+1. Configure the OpenTelemetry [Host Metrics Receiver](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/hostmetricsreceiver):
```
You can link this page as well - https://signoz.io/docs/userguide/hostmetrics/

makeavish commented on data/docs/migration/migrate-from-newrelic.mdx:
```diff
@@ -0,0 +1,29 @@
+
+Please refer to the following guides for step-by-step instructions on migrating each component from New Relic to SigNoz:
+
+- [Migrating Dashboards](/docs/migration/migrate-from-newrelic/dashboards) - Convert your New Relic dashboards
```
Update order according to ordering on sidebar.

makeavish commented on data/docs/migration/migrate-from-newrelic/traces.mdx:
```diff
@@ -0,0 +1,130 @@
+| `entity.type`                 | `service.namespace`                |
+| `transactionName`             | `http.route`                       |
+| `http.method`                 | `http.method`                      |
+| `http.url`                    | `http.url`                         |
```
In latest otel spec this has been updated to url.full and even other attribute names seems incorrect please check latest otel specs and update
Ref: https://opentelemetry.io/docs/specs/semconv/http/http-spans/
https://opentelemetry.io/docs/specs/semconv/database/database-spans/

makeavish commented on data/docs/migration/migrate-from-newrelic/logs.mdx:
```diff
@@ -0,0 +1,194 @@
+
+### 2. Using FluentBit with OpenTelemetry
+
+If you're already using FluentBit with Loki, you can reconfigure it to work with OpenTelemetry. First, configure FluentBit to forward logs to the OpenTelemetry Collector:
```
Loki?

makeavish reviewed (commented):
Please mention about UI based SigNoz query builder as well.
No mention of New Relic Synthetics monitoring migration - SigNoz only supports http check metrics which can be linked here - https://signoz-web-git-fork-tdabasinskas-migratenewrelic-signoz.vercel.app/docs/monitor-http-endpoints/
Infrastructure integrations (AWS, Azure, GCP) migration not covered - SigNoz has one click integration for AWS, others are coming soon.
Mention cost savings https://signoz.io/product-comparison/signoz-vs-newrelic/ 
https://signoz.io/blog/pricing-comparison-signoz-vs-datadog-vs-newrelic-vs-grafana/

tdabasinskas committed:
Update data/docs/migration/migrate-from-newrelic/alerts.mdx

Co-authored-by: Chitransh <31571545+Calm-Rock@users.noreply.github.com>

tdabasinskas committed:
Apply suggestions from code review

Co-authored-by: Chitransh <31571545+Calm-Rock@users.noreply.github.com>

tdabasinskas committed:
fix(migrate-from-newrelic): update the paragraph on Terraform provider

tdabasinskas committed:
chore(migrate-from-newrelic): update the guides order

tdabasinskas committed:
chore(migrate-from-newrelic): update the link to hostmetrics

tdabasinskas committed:
fix(migrate-from-newrelic): update the paragraph on FluentBit

tdabasinskas committed:
chore(migrate-from-newrelic): update trace attribute mapping table

tdabasinskas committed:
feat(migrate-from-newrelic): add additional information about infrastructure module

tdabasinskas committed:
Merge branch 'main' into migrate_newrelic

tdabasinskas committed:
chore(migrate-from-newrelic): fix spacing in the navigation YAML

tdabasinskas committed:
feat(migrate-from-newrelic): add section on Query Builder

tdabasinskas committed:
feat(migrate-from-newrelic): add section on synthetic monitoring

tdabasinskas committed:
chore(migrate-from-newrelic): update the tiles for metrics migration approaches

tdabasinskas committed:
feat(migrate-from-newrelic): add paragraph on cloud integration

tdabasinskas committed:
feat(migrate-from-newrelic): add details on costs

tdabasinskas commented:
Thanks for the review!

> Please mention about UI based SigNoz query builder as well. No mention of New Relic 

Added.

> Synthetics monitoring migration - SigNoz only supports http check metrics which can be linked here - https://signoz-web-git-fork-tdabasinskas-migratenewrelic-signoz.vercel.app/docs/monitor-http-endpoints/

Nice! Added a section on that.

> Infrastructure integrations (AWS, Azure, GCP) migration not covered - SigNoz has one click integration for AWS, others are coming soon.

Added, but let me know if it makes sense. I'm not sure where to link on this. I couldn't find any docs on one-click integration with AWS. These doesn't seem to be entirely that:

- https://signoz.io/guides/aws-monitoring/
- https://signoz.io/docs/ec2-monitoring/
- https://signoz.io/docs/userguide/send-cloudwatch-logs-to-signoz/

> Mention cost savings https://signoz.io/product-comparison/signoz-vs-newrelic/ https://signoz.io/blog/pricing-comparison-signoz-vs-datadog-vs-newrelic-vs-grafana/

Added to the main page.

tdabasinskas committed:
chore(feed): rebuild

tdabasinskas committed:
chore(nav): add missing comma

tdabasinskas committed:
fix(feed): remove unnecessary data

makeavish commented:
> Added, but let me know if it makes sense. I'm not sure where to link on this. I couldn't find any docs on one-click integration with AWS. These doesn't seem to be entirely that:
Ah, it seems we still haven't shipped docs on this, but we have added around support for 12 AWS services with 1-click integration already
https://signoz.io/blog/native-aws-integrations-with-autodiscovery/
